File: crates/domain-scan-core/tests/fixtures/csharp/interfaces.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace DomainScan.Tests
{
    public interface IUserRepository
    {
        Task<User> GetById(int id);
        Task<List<User>> GetAll();
        Task Save(User user);
        void Delete(int id);
    }

    public interface INotificationService
    {
        void SendEmail(string to, string subject, string body);
        Task SendSmsAsync(string phoneNumber, string message);
    }

    public interface IRepository<T> where T : class
    {
        Task<T> FindById(int id);
        Task<IEnumerable<T>> FindAll();
        Task Add(T entity);
        Task Update(T entity);
        void Remove(T entity);
    }

    internal interface IInternalService
    {
        void DoWork();
    }
}

File: crates/domain-scan-core/tests/fixtures/php/interfaces.php
<?php

interface UserRepositoryInterface {
    public function findById(int $id): ?User;
    public function save(User $user): void;
    public function delete(int $id): void;
}

interface NotificationServiceInterface extends ServiceInterface {
    public function send(string $message): bool;
    public function isConnected(): bool;
}

interface CacheInterface {
    public function get(string $key): mixed;
    public function set(string $key, mixed $value, int $ttl = 3600): void;
    public function has(string $key): bool;
}
